### ISSUE TITLE
Suggestion: Turn on "Require WiFi when downloading" by default

### ISSUE THREAD

**Issue by Whoever4976** [jmshrv/finamp#981]
I was recently quite annoyed when the app used up my entire mobile data quota for the month in a singular day while attempting to finish a 44 GB download in the background which i had started a couple of hours earlier while still inside my home WiFi-network.
I know it is not a written rule to design mobile apps with safeguards against excessive data usage but i do feel like this is something that most users expect from any app they install on their phone. I only noticed the relevant setting afterwards which, by default, allows for downloads of any(!) size using mobile data.   

I know that this is also a user error on my part because i haven't checked the relevant setting to be sure, but quite frankly, it just didn't cross my mind that any modern app would allow for such behaviour and i don't think that i am in the minority with this assumption.

Luckily i was using a Prepaid SIM card and the cost was limited to a few euros and my quota just ran out. But if i had been using another contract, downloading 44GB on mobile data could have been quite an expensive mistake which could have been avoided if i checked the setting beforehand **or** if the app had, in my opinion, better defaults

So i suggest that, at the very least, the setting "Require WiFi when downloading" is turned on by default. An additional feature could be a prompt which appears when a download is pending but no wifi is connected. Something like "No Wifi connected. Do you want to use mobile data to continue the download anyway?"

**Comment by Chaphasilor** [comment]
I agree, will update the default! Sorry about the rogue download :(  
For now, there won't be any notification or size limit, since that's a bit more complex to implement...

**Comment by flloschy** [comment]
Uhh yeah we talked about it but I dont think I implemented it for a reason I cant recall... Let me check

**Comment by flloschy** [comment]
Im unsure on how to trigger the notification though, preferably id like to get an update from the downloader package when its paused but that doesnt seam to be get-able *(does the downloader stop and resume? I tried to understand the code but there isnt really a pause&resume I could find in relation to wifi state inside the package?)*
Alternatively I could add another listener to the current Connectivity listener and check if the download queue is empty or not but that doesnt feel quite right

**Comment by flloschy** [comment]
I ended up going the second route: https://github.com/jmshrv/finamp/pull/1162/commits/cbe804e6859030618097f1c6015d0c2cdc41915b
